extensions/wikia/Blogs/src/Hooks/UserCan/Check/BlogArticleCheckFactory.php

```diff
@@ -0,0 +1,34 @@
+<?php
+namespace Extensions\Wikia\Blogs\Hooks\UserCan\Check;
+
+use Extensions\Wikia\Blogs\DependencyFactory;
+use Extensions\Wikia\Blogs\Hooks\UserCan\Action;
+
+class BlogArticleCheckFactory {
+	/** @var DependencyFactory $dependencyFactory */
+	private $dependencyFactory;
+
+	public function __construct( DependencyFactory $factory ) {
+		$this->dependencyFactory = $factory;
+	}
+
+	public function newActionCheck( string $action ) {
+		switch ( $action ) {
+			case Action::CREATE:
+				return new BlogArticleCreateActionCheck( $this->dependencyFactory );
+				break;
+			case Action::EDIT:
+				return new BlogArticleEditActionCheck( $this->dependencyFactory );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the code after this "return". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1763)